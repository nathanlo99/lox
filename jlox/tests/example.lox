
for (var i = 1; i < 5; i = i + 1) {
  print(i * i);
}

class Duck {
  init(name) {
    this.name = name;
  }

  quack() {
    print(this.name + " quacks");
  }
}

var duck = Duck("Waddles");
duck.quack();

fun make_adder(n) {
  fun adder(i) {
    return n + i;
  }
  return adder;
}
var add5 = make_adder(5);
print(add5(1));
print(add5(100));

class Circle {
  init(radius) {
    this.radius = radius;
  }

  area {
    return 3.141592653 * this.radius * this.radius;
  }
}

var circle = Circle(4);
print(circle.area); // Prints roughly "50.2655".
