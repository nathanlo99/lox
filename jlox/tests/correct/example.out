1
4
9
16
Waddles quacks
6
105
